mypy/messages.py

```diff
@@ -256,16 +256,28 @@ def span_from_context(ctx: Context) -> Iterable[int]:
             assert origin_span is not None
             origin_span = itertools.chain(origin_span, span_from_context(secondary_context))
 
+        column = context.column if context else -1
+        end_line = context.end_line if context else -1
+        end_column = context.end_column if context else -1
+
+        # set end line and column to same as start of context for function definitions
+        # this avoids errors being reported in IDEs for the whole function
+        # TODO: figure out if it's possible to find the end of the function definition line
```

Comment:
One approach for this could be to add a new attribute to the `FuncDef` class and set it when we're creating the FuncDef object in `fastparse.py`.